cyipopt/scipy_interface.py

```diff
@@ -533,3 +538,39 @@ def minimize_ipopt(fun,
                           nfev=problem.nfev,
                           njev=problem.njev,
                           nit=problem.nit)
+
+def _minimize_ipopt_iv(fun, x0, args, kwargs, method, jac, hess, hessp,
+                       bounds, constraints, tol, callback, options):
+    # basic input validation for minimize_ipopt
+    if fun is not None and not callable(fun):
+        raise ValueError('`fun` must be callable.')
+    x0 = np.asarray(x0)[()]
+    if not np.issubdtype(x0.dtype, np.number):
+        raise ValueError('`x0` must be a numeric array.')
+    if not np.iterable(args):
+        args = (args,)
+    kwargs = dict() if kwargs is None else kwargs
+    if not isinstance(kwargs, dict):
+        raise ValueError('`kwargs` must be a dictionary.')
+    if method is not None:  # this will be updated when gh-200 is merged
+        raise NotImplementedError('`method` is not yet supported.`')
+    if jac is not None and jac not in {True, False} and not callable(jac):
+        raise ValueError('`jac` must be callable or boolean.')
+    if hess is not None and not callable(hess):
+        raise ValueError('`hess` must be callable.')
+    if hessp is not None:
+        raise NotImplementedError('`hessp` is not yet supported by Ipopt.`')
```

Comment:
Previously there was some input validation of `jac` and `hessp` in `IpoptProblemWrapper`, but this will happen first. If `IpoptProblemWrapper` is only used by `minimize_ipopt`, shall I remove the (now redundant) input validation of these things there?

Reply:
Given that IpoptProblemWrapper is an independent class, any input validation for it belongs there. So my recommendation is to validate everything you can there and then whatever is left could be validated here.

Reply:
> I couldn't tell - is it considered public?

Everything is public unless it has been explicitly labeled as not public.

Reply:
Ok, I'll mov what can be moved.

Reply:
Done.